### PR TITLE
update google maps directive to handle setup via other directives

### PR DESCRIPTION
**Pull request by tazo90**
This update allow to change configuration of the maps from `lux.google` via other directives e.g. `bmll.exchanges.map`

- Ref to: [bmll-portal: #112](https://github.com/bmlltech/bmll-portal/issues/112)

### CODE REVIEW

**Comment by lsbardel** [comment]
@reupen , @jeremyherr can you review this pull request?

File: luxjs/google/module.js [hunk 1]
```diff
@@ -37,13 +37,39 @@
...
+                            var gmarker = new google.maps.Marker(angular.extend({
+                                position: location,
+                                map: map,
+                            }, marker));
+                            return gmarker;
+                        }
+                    };
+
+                    // Add marker using lat and lng
+                    self.addLocation = function(map, marker, lat, lng) {
```
Comment (reupen) [inline luxjs/google/module.js]: We should have some consistency with how functions are named, so this should be something like addMarkerByLatLng

File: luxjs/google/module.js [hunk 2]
```diff
@@ -37,13 +37,39 @@
...
+                        var loc = new google.maps.LatLng(lat, lng);
+                        return self.addMarker(map, marker, loc);
+                    };
+
+                    // Returns an instance of location for specific lat and lng
+                    self.getLocation = function(lat, lng) {
```
Comment (reupen) [inline luxjs/google/module.js]: This should be createLocation as it is creating an object

File: luxjs/google/module.js [hunk 3]
```diff
@@ -37,13 +37,39 @@
...
+                        return new google.maps.LatLng(lat, lng);
+                    };
+
+                    // Returns an instance of InfoWindow for specific content
+                    self.getInfoWindow = function(content) {
```
Comment (reupen) [inline luxjs/google/module.js]: This should be createInfoWindow as it is creating an object

File: luxjs/google/module.js [hunk 4]
```diff
@@ -55,20 +81,32 @@
...
+                link: function (scope, element, attrs, controller) {
+                    if(!scope.googlemaps) {
+                        $lux.log.error('Google maps url not available. Cannot load google maps directive');
+                        return;
+                    }
+                    require([scope.googlemaps], function () {
+                        controller.initialize(scope, element, attrs);
+
+                        // Setup map by another directive
+                        if (scope.hasOwnProperty('setup'))
```
Comment (jeremyherr) [inline luxjs/google/module.js]: Are you just checking that scope.setup() is defined? If so then it's better to say
```if (typeof scope.setup === 'function')```
Or do you actually need to make sure you are using this particular scope's setup method, and do not want to use an inherited one? That seems unlikely, but I am not familiar with this code.

**Comment by tazo90** [comment]
Guys thanks for the comments. I will do suggested changes.

**Comment by tazo90** [comment]
I did updates for this ticket.

**Pull request merged**